Cloud Application Manager/Automating Deployments/running-code-outside-an-instance.md

```diff
@@ -0,0 +1,99 @@
+
+### Availability and visibility
+
+Task variables are only available for public, versioned, verified boxes. Because of this verification process, there is no Task Variable option in the code section of a box: you should create a File variable and then send a publish request. The variables will be marked as *private* and will not appear in the lifecycle editor. 
+
+### Example
+
+This example will guide you through the steps of creating a box, adding a script, publishing it with the script as a task variable and running it.
+
+* First, create a [script box](../Core Concepts/boxes.md) as you would normally do.
+* Go to the **Code** tab, click on New, then click on the **File** tab.
+* Enter a descriptive name for the variable and click on *Create Empty File**. You may enter a file name with the `sh` suffix, although it's not strictly necessary.
+![new file variable](../../images/cloud-application-manager/task-variables-01.png)
+* The variable will appear on the list. Click on the file name and an editor will appear.
+![file variable ready](../../images/cloud-application-manager/task-variables-02.png)
+* Enter the code you want to run. It should start with the `#!/bin/bash` sentence. After that, you may use bash, nodejs or python. You may embed code as *HEREDOCs* and then run them. Other box variables are accessible via Jinja [syntax](syntax-for-variables.md). For example, the code below prints the value of *variable1*. Once ready, save the code.
+![code editor](../../images/cloud-application-manager/task-variables-03.png)
+
+**The code used in the example is listed here for reference.**
+```sh
+#!/bin/bash
+
+echo "printing from bash"
+
+pythonfile=/tmp/pytnontest.py
+nodefile=/tmp/nodetest.js
+
+cat <<PYTHON> $pythonfile
+#!/usr/bin/python
+print("printing from python")
+
+PYTHON
+
+chmod +x $pythonfile
+$pythonfile
+
+cat <<NODE> $nodefile
+console.log('printing from node the value of variable1: {{ variable1 }}')
+
+NODE
+
+node $nodefile
+
+elasticbox set variable1 value_set_from_task
+
+```
+
+* Move to the **Versions** tab, then click on the **New** button. Add a description and click on Save.
+![new version](../../images/cloud-application-manager/task-variables-04.png)
+* The version you just added will appear in the list. Click the gear button on the right end of the row, then click on **Publish Box**.
+![publish box](../../images/cloud-application-manager/task-variables-05.png)
+* Enter the details of the box as you want them to appear on the catalog. Specify the File variables you want to convert into Task variables.
+![publish details](../../images/cloud-application-manager/task-variables-06.png)
+* Your box will undergo a verification process and, if approved, will eventually be published and appear in the catalog. When you access it, a green checkmark will appear close to the version you published.
```

Comment:
Please, mention that the user should be notified about the result of the review.